scalyr_agent/scalyr_monitor.py

```diff
@@ -250,6 +256,33 @@ def config_from_monitors(self, manager):
         """
         pass
 
+    def _get_sleep_delay(self):
```

Comment:
Ok, I thought this was just being added to the first sleep time, but now I see that it is being added on every sleep.  Yeah, that makes it more invasive and could hurt some of the dashboards.  (Some of the dashboards based on metrics emitted by the monitors do things like calculate the average value over a minute interval... if we keep do this, then in some intervals it will get two samples vs none in others.)

I think it would be sufficient to just delay the sleep for the first sample.  I.e, right after the agent starts, instead of having all of the monitors start collecting right then, we jitter each one so their start time is spread out over a 60 sec interval.  After that, they each sleep the normal sample interval.  So, the frequency is still fixed at 60 seconds but the phase is offset for all of the monitors.

What do you think of that?

Reply:
Yeah, I think this should also work as long as we do it on on monitor init aka when monitors are started / restarted.

Reply:
Alright, I pushed a change so we only sleep before first sample gather interval and after that we used fixed interval - 09d5e4a34797f8c5576deeb3ede18f0d810f94f7. 

I thought that adding random jitter to each sleep should work as well (as long as it's smaller than minimum roll up / aggregation window we use on the server side), since server should already take such things (time drifts, delayed metric, missed samples, etc.) into account, but yeah, this approach is probably indeed safer.

And since we only use this sleep before first sample gather interval, I'm also fine with enabling this option by default - it should have no negative impact or consequences.

@czerwingithub  - WDYT about enabling it by default? I would still leave config option / flag there to disable it, if needed.